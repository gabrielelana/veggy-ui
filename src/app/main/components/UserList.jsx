const UserList = ({users, onToggleUser}) => {
  const userRows = users.map(u => (
      <UserRow key={u.user_id} 
        onSelect={() => onToggleUser(u.user_id)} 
        selected={u.selected} 
        username={u.username} />
      ))
    
  return (
    <div className="column">
      <nav className="panel" style={{backgroundColor: '#FFFFFF'}}>
        <p className="panel-heading">
          Users
        </p>
        {userRows}
      </nav>
    </div>
  )
}

const UserRow = ({username, selected, onSelect}) => {
  const rowStyle = selected?'selectedRow':''
  
  return (
    <a className={`${rowStyle} panel-block`} href="#" onClick={onSelect} >
      <span className="panel-icon">
        <i className="fa fa-user"></i>
      </span>
      {username}
    </a>)
}

export default UserList