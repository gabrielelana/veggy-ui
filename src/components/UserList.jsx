const React = require('react')

const UserList = React.createClass({
  render(){
    return (
      <div className="column">
        <nav className="panel">
          <p className="panel-heading">
            Users
          </p>
          <a className="panel-block is-active" href="#">
            <span className="panel-icon">
              <i className="fa fa-user"></i>
            </span>
            user 1
          </a>
          <a className="panel-block" href="#">
            user 2
            <span className="panel-icon">
              <i className="fa fa-user"></i>
            </span>
          </a>
        </nav>
      </div>
      )
  }
})

export default UserList