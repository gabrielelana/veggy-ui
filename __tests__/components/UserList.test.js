import React from 'react'
import renderer from 'react-test-renderer'
import UserList from '../../src/app/main/components/UserList'

describe('<UserList />', () => {
  
  it('should display the tasks', () => {
    const users = [{
      userId: 1,
      selected: false,
      username: 'ema'
    },{
      userId: 2,
      selected: true,
      username: 'gabriele'
    }]
    const component = renderer.create(<UserList users={users} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})