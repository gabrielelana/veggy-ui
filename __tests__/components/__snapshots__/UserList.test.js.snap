// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> should display the tasks 1`] = `
<div
  className="column"
>
  <nav
    className="panel"
    style={
      Object {
        "backgroundColor": "#FFFFFF",
      }
    }
  >
    <p
      className="panel-heading"
    >
      Users
    </p>
    <a
      className=" panel-block"
      href="#"
      onClick={[Function]}
    >
      <span
        className="panel-icon"
      >
        <i
          className="fa fa-user"
        />
      </span>
      ema
    </a>
    <a
      className="selectedRow panel-block"
      href="#"
      onClick={[Function]}
    >
      <span
        className="panel-icon"
      >
        <i
          className="fa fa-user"
        />
      </span>
      gabriele
    </a>
  </nav>
</div>
`;
